cpp/oneapi/dal/algo/pca/backend/gpu/infer_kernel_dpc.cpp

```diff
@@ -39,25 +39,65 @@ static result_t infer(const context_gpu& ctx, const descriptor_t& desc, const in
     dal::detail::check_mul_overflow(row_count, component_count);
 
     const auto data_nd = pr::table2ndarray<Float>(queue, data, sycl::usm::alloc::device);
+    auto means_nd = pr::table2ndarray_1d<Float>(queue, means, sycl::usm::alloc::device);
+    auto mean_centered_data_nd =
+        pr::ndarray<Float, 2>::empty(queue, { row_count, col_count }, sycl::usm::alloc::device);
     const auto eigenvectors_nd =
         pr::table2ndarray<Float>(queue, eigenvectors, sycl::usm::alloc::device);
 
+    pr::elementwise_difference(queue, row_count, data_nd, means_nd, mean_centered_data_nd);
+
     auto res_nd = pr::ndarray<Float, 2>::empty(queue,
                                                { row_count, component_count },
                                                sycl::usm::alloc::device);
     sycl::event gemm_event;
     {
         ONEDAL_PROFILER_TASK(gemm, queue);
-        gemm_event = pr::gemm(queue, data_nd, eigenvectors_nd.t(), res_nd, Float(1.0), Float(0.0));
+        gemm_event = pr::gemm(queue,
+                              mean_centered_data_nd,
+                              eigenvectors_nd.t(),
+                              res_nd,
+                              Float(1.0),
+                              Float(0.0));
     }
-    const auto res_array = res_nd.flatten(queue, { gemm_event });
-    auto res_table = homogen_table::wrap(res_array, row_count, component_count);
 
+    if (eigenvalues.has_data()) {
+        auto eigenvalues_nd =
+            pr::table2ndarray_1d<Float>(queue, eigenvalues, sycl::usm::alloc::device);
+        auto sqrt_eigenvalues_nd =
+            pr::ndarray<Float, 1>::empty(queue, { component_count }, sycl::usm::alloc::device);
+        pr::elementwise_sqrt(queue, eigenvalues_nd, sqrt_eigenvalues_nd);
+
+        auto result_whitened_nd = pr::ndarray<Float, 2>::empty(queue,
+                                                               { row_count, component_count },
+                                                               sycl::usm::alloc::device);
+        sycl::event whiten_event;
+        {
+            ONEDAL_PROFILER_TASK(elementwise_division, queue);
+            whiten_event = pr::elementwise_division(queue,
+                                                    row_count,
+                                                    res_nd,
+                                                    sqrt_eigenvalues_nd,
+                                                    result_whitened_nd);
+        }
+
+        const auto res_array_whitened = result_whitened_nd.flatten(queue, { whiten_event });
+        const auto res_table = homogen_table::wrap(res_array_whitened, row_count, component_count);
+
+        return result_t{}.set_transformed_data(res_table);
+    }
+
+    const auto res_array = res_nd.flatten(queue, { gemm_event });
+    const auto res_table = homogen_table::wrap(res_array, row_count, component_count);
     return result_t{}.set_transformed_data(res_table);
 }
 
```

Comment:
it seems to me that our bazel tests almost do not cover infer for pca, might be a good time to add them